[Style] Don't require specifying an animation wrapper in standard cases

weinig opened the pull request:
#### a56042534dc01d55c8ae12d269f6690c88565135
<pre>
[Style] Don&apos;t require specifying an animation wrapper in standard cases
<a href="https://bugs.webkit.org/show_bug.cgi?id=303068">https://bugs.webkit.org/show_bug.cgi?id=303068</a>

Reviewed by Antoine Quint.

Now that almost all properties use one of the standard animation wrappers,
StyleTypeWrapper, DiscreteWrapper, VisitedAffectedStyleTypeWrapper,
CoordinatedValueListPropertyStyleTypeWrapper, and DiscreteCoordinatedValueListPropertyWrapper,
we can remove the requirement that they be specified explicitly in most cases.

The rule is:

    - if &quot;animation-wrapper&quot; is specified, use what it specifies
    - otherwise
        - if &quot;animation-type&quot; is &quot;discrete&quot;
            - if its a coordinated list value use `DiscreteCoordinatedValueListPropertyWrapper`
            - otherwise, use `DiscreteWrapper`
        - otherwise:
            - if its a coordinated list value use `CoordinatedValueListPropertyStyleTypeWrapper`
            - if its a visited color property use `VisitedAffectedStyleTypeWrapper`
            - otherwise, use `StyleTypeWrapper`

* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/scripts/process-css-properties.py:
* Source/WebCore/css/scripts/test/TestCSSPropertiesResults/StyleInterpolationWrapperMap.cpp:

Canonical link: <a href="https://commits.webkit.org/303547@main">https://commits.webkit.org/303547@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f3b08502d4a64b5c5fb1c96e6b0323e264a9eca1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/132676 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/5171 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/43747 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/140204 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/84703 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/134546 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/5498 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/4930 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/101433 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/68730 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/135622 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/3871 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/118857 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/82226 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/3757 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/1434 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/83438 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/112722 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/36971 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/142859 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/4841 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/37560 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/109809 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/4923 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/4184 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109986 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27902 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/3692 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/115128 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/58315 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/4895 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/33476 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/164151 "Build is in progress. Recent messages:") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/4733 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/68346 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/35/builds/164151 "Build is in progress. Recent messages:") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/4986 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/4852 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

graouts reviewed (approved):
Great stuff! So we've reached a point where `CSSProperties.json` can mostly be descriptive and adding a new run-of-the-mill CSS property won't require any animation-specific code?

weinig committed:
[Style] Don't require specifying an animation wrapper in standard cases
https://bugs.webkit.org/show_bug.cgi?id=303068

Reviewed by Antoine Quint.

Now that almost all properties use one of the standard animation wrappers,
StyleTypeWrapper, DiscreteWrapper, VisitedAffectedStyleTypeWrapper,
CoordinatedValueListPropertyStyleTypeWrapper, and DiscreteCoordinatedValueListPropertyWrapper,
we can remove the requirement that they be specified explicitly in most cases.

The rule is:

    - if "animation-wrapper" is specified, use what it specifies
    - otherwise
        - if "animation-type" is "discrete"
            - if its a coordinated list value use `DiscreteCoordinatedValueListPropertyWrapper`
            - otherwise, use `DiscreteWrapper`
        - otherwise:
            - if its a coordinated list value use `CoordinatedValueListPropertyStyleTypeWrapper`
            - if its a visited color property use `VisitedAffectedStyleTypeWrapper`
            - otherwise, use `StyleTypeWrapper`

* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/scripts/process-css-properties.py:
* Source/WebCore/css/scripts/test/TestCSSPropertiesResults/StyleInterpolationWrapperMap.cpp:

Canonical link: https://commits.webkit.org/303547@main
